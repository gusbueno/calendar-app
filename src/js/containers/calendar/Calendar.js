import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import moment from 'moment'

import styles from './Calendar.scss'
import Month from '../../components/month/Month'
import { onOpenEventDetail } from '../../actions/eventDetail'

class Calendar extends PureComponent {
  constructor (props) {
    super(props)
    this.state = {
      year: null,
      month: null
    }

    this._previousMonth = this._previousMonth.bind(this)
    this._nextMonth = this._nextMonth.bind(this)
  }

  componentWillMount () {
    const year = Number(moment().format('YYYY'))
    const month = Number(moment().format('M'))
    this.setState({ year, month })
  }

  _previousMonth () {
    const previousDate = moment(`${this.state.month}-${this.state.year}`, ['M-YYYY']).subtract(1, 'months')
    const year = Number(previousDate.format('YYYY'))
    const month = Number(previousDate.format('M'))
    this.setState({ year, month })
  }

  _nextMonth () {
    const nextDate = moment(`${this.state.month}-${this.state.year}`, ['M-YYYY']).add(1, 'months')
    const year = Number(nextDate.format('YYYY'))
    const month = Number(nextDate.format('M'))
    this.setState({ year, month })
  }

  render () {
    const monthName = moment(this.state.month, ['MM']).format('MMMM')
    const daysInMonth = moment(`${this.state.year}-${this.state.month}`, ['YYYY-MM']).daysInMonth()
    return (
      <div className={styles.calendar}>
        <div className={styles['calendar-header']}>
          <div className={styles['previous-btn']} onClick={this._previousMonth}>
            <span>{'<'}</span>
          </div>
          <div className={styles.date}>
            <span className={styles.year}>{this.state.year}</span>
            <span className={styles.month}>{monthName}</span>
          </div>
          <div className={styles['next-btn']} onClick={this._nextMonth}>
            <span>{'>'}</span>
          </div>
        </div>
        <Month year={this.state.year} month={this.state.month} daysInMonth={daysInMonth} onAddEvent={this.props.onAddEvent} />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return ({
    onAddEvent: (data) => {
      dispatch(onOpenEventDetail(true, data))
    }
  })
}

Calendar.propTypes = {
  onAddEvent: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(Calendar)
