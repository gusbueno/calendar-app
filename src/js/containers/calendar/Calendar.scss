@import "../../../styles/variables";

.calendar {
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  .calendar-header {
    display: flex;
    align-items: center;

    .previous-btn, .next-btn {
      padding: 10px;
      box-sizing: border-box;
      border: 1px solid $color-green;
      background-color: $color-green;
      border-radius: 10px;
      &:hover {
        cursor: pointer;
      }

      span {
        font-family: 'Roboto-Bold';
        font-size: 30px;
        color: $color-bone;
      }
    }

    .date {
      padding: 10px;
      box-sizing: border-box;
      display: flex;
      align-items: center;

      .year {
        font-family: 'Roboto-bold';
        font-size: 38px;
        margin-right: 10px;
        color: $color-black;
      }
  
      .month {
        font-family: 'Roboto-Regular';
        font-size: 30px;
        color: $color-green;
      }
    }
  }
}