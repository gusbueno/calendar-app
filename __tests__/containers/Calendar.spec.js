
/* global describe, it, expect */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import Calendar from '../../src/js/containers/calendar/Calendar'

const storeFake = {
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => {
    return {
      calendar: {
        events: []
      }
    }
  }
}

describe('<Calendar /> container', () => {
  it('should match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={storeFake}>
        <Calendar />
      </Provider>
    )
    const json = tree.toJSON()
    expect(json).toMatchSnapshot()
  })

  const calendarShallow = shallow(
    <Provider store={storeFake}>
      <Calendar />
    </Provider>
  )

  it('should exists', () => {
    expect(calendarShallow.length).toEqual(1)
  })
})
