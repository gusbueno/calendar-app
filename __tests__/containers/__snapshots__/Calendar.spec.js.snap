// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> container should match with the snapshot 1`] = `
<div
  className="calendar"
>
  <div
    className="calendar-header"
  >
    <div
      className="previous-btn"
      onClick={[Function]}
    >
      <span>
        &lt;
      </span>
    </div>
    <div
      className="date"
    >
      <span
        className="year"
      >
        2018
      </span>
      <span
        className="month"
      >
        May
      </span>
    </div>
    <div
      className="next-btn"
      onClick={[Function]}
    >
      <span>
        &gt;
      </span>
    </div>
  </div>
  <div
    className="month"
  >
    <div
      className="month-header"
    >
      <div
        className="week-day"
      >
        <span>
          Sunday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Monday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Tuesday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Wednesday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Thursday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Friday
        </span>
      </div>
      <div
        className="week-day"
      >
        <span>
          Saturday
        </span>
      </div>
    </div>
    <div
      className="month-days"
    >
      <div
        className="day"
      />
      <div
        className="day"
      />
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            1
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            2
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            3
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            4
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            5
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            6
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            7
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            8
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            9
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            10
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            11
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            12
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            13
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            14
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            15
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            16
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            17
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            18
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            19
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            20
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            21
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            22
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            23
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            24
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            25
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            26
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            27
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            28
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            29
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            30
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
      <div
        className="day"
      >
        <div
          className="day-header"
        >
          <span>
            31
          </span>
          <div
            className="add-event-btn"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
